src/utils/documentProducer/variables.ts

```diff
@@ -112,15 +113,24 @@ export const isValueEmpty = (
   }
 };
 
-export const missingRequiredFields = (variablesWithValues: VariableWithValues[]) => {
+export const missingRequiredFields = (
+  variablesWithValues: VariableWithValues[],
+  pendingTableValues?: Map<number, VariableTableCell[][]>
+) => {
   const allRequiredVariables = variablesWithValues.filter(
     (v) => v.isRequired && variableDependencyMet(v, variablesWithValues)
   );
-
   const missingRequiredFields = allRequiredVariables.some((variable) => {
     const values = variable.values;
```

Comment:
I think this may work and we can move it to the top of the function. 

```
if (variable.type === 'Table') {
  const tableCells = !pendingTableValues?.get(variable.id)

  // Return true if there is no row
  if (!tableCells || tableCells.length === 0) {
    return true;
  }
  
  // Return true if every cell is empty
  return tableCells.every((row) => row.every(cell) => {
    if (!cell.values || cell.values.length ===0) {
        return true;
    } 
    return cellValue(cell.values[0]).toString() === ''
  }))
}
```

Reply:
Ahh I understand.  Quite tricky because I think we need to catch the case of not adding a row to an empty table, as well as not modifying an existing table.

Reply:
Perhaps we can add a `stagedCellValues` similar to `stagedVariableValues`

Reply:
Yes! That was my first idea but was getting trouble with typescript

Reply:
But if there is no row, then values of the table will be empty, so it will be marked as required in the first check